internal/servers/service_rotate_roots.go

```diff
@@ -0,0 +1,24 @@
+package servers
+
+import (
+	"context"
+
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/nodeenrollment"
+	"github.com/hashicorp/nodeenrollment/rotation"
+)
+
+// RotateRoots is a domain service function that initiates a rotation of root certificates
+// via a call to the nodenenrollment RotateRootCertificates function
+// Accepts the nodeenrollment option, WithCertificateLifetime(time.Duration) to specify the lifetime
+// of the generated cert(s)
+func RotateRoots(ctx context.Context, workerAuthRepo *WorkerAuthRepositoryStorage, opt ...nodeenrollment.Option) error {
+	const op = "servers.RotateRoots"
+
```

Comment:
perhaps check workAuthRepo != nil ?  I realized RotateRootCertificates also checks this, but why make the call with an invalid parameter?